File: syverro-web/src/types/book.ts
// src/types/book.ts

export type BookStatus = 
  | 'reading'
  | 'rereading'
  | 'completed'
  | 'want_to_read'
  | 'postponed'
  | 'abandoned';

export interface Quote {
  id: string;
  text: string;
  page: number | null;
  note: string | null;
  createdAt: number;
}

export interface PersonalBook {
  userId: string;
  bookId: string;
  status: BookStatus;
  currentPage: number;
  rating: number | null;
  favorite: boolean;
  notes: string;
  quotes: Quote[];
  addedAt: number;
  updatedAt: number;
}

export interface GlobalBook {
  id: string;
  title: string;
  author: string;
  cover: string | null;
  genres: string[];
  totalPages: number;
  authorCountry: string | null;
  originalYear: number | null;
  description: string | null;
  averageRating: number | null;
  totalRatings: number;
  createdAt: number;
  subtitle?: string;
  authors?: string[];
  originalLanguage?: string;
  subgenres?: string[];
  mood?: string[];
  vibe?: string[];
  themes?: string[];
  motifs?: string[];
  culturalTradition?: string;
  historicalPeriodWritten?: string;
  historicalPeriodSetting?: string;
  plotLinearity?: 'linear' | 'nonlinear';
  povCount?: number;
  textComplexity?: 'low' | 'medium' | 'high';
  narrativePace?: 'slow' | 'medium' | 'fast';
  descriptionDensity?: 'low' | 'medium' | 'high';
  dialogueRatio?: number;
  isClassic?: boolean;
  isNiche?: boolean;
  moderationStatus?: 'pending' | 'approved' | 'rejected';
  moderationReason?: string;
  triggerWarnings?: string[];
}

export interface EnrichedBook extends GlobalBook {
  personal: PersonalBook | null;
}

export type NewGlobalBook = Omit<GlobalBook, 'id' | 'createdAt' | 'averageRating' | 'totalRatings'>;

export interface Edition {
  id: string;
  bookId: string;
  publisher: string;
  year: number;
  country: string;
  translator?: string;
  editor?: string;
  series?: string;
  seriesPosition?: number;
  format: 'hardcover' | 'paperback' | 'ebook' | 'audiobook';
  pages: number;
  language: string;
  cover: string | null;
  isbn10?: string;
  isbn13?: string;
}

export interface BookVibe {
  bookId: string;
  mood: string[];
  vibe: string[];
  themes: string[];
  motifs: string[];
  source: 'manual' | 'ai' | 'crowd';
  confidence: number;
}

export interface Character {
  id: string;
  bookId: string;
  name: string;
  isMain: boolean;
  archetype?: string;
  description?: string;
}

export interface Review {
  id: string;
  bookId: string;
  userId: string;
  title: string | null;
  content: string;
  rating: number | null;
  isSpoiler: boolean;
  moderationStatus: 'pending' | 'approved' | 'rejected';
  helpfulCount: number;
  createdAt: number;
  updatedAt: number | null;
}

// ============================================
// EDIT PROPOSAL — модерация изменений GBE
// ============================================
export type EditProposalStatus = 'pending' | 'approved' | 'rejected';

export interface EditProposal {
  id: string;
  bookId: string;
  userId: string;
  changedFields: Partial<GlobalBook>;
  reason?: string;
  status: EditProposalStatus;
  createdAt: number;
  reviewedAt?: number;
  moderatorId?: string;
  moderatorComment?: string;
}

File: syverro-web/src/pages/book.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID

class BookCreate(BaseModel):
    title: str
    author: str
    status: Optional[str] = "planned"
    rating: Optional[int] = None
    cover: Optional[str] = None
    section: Optional[str] = None
    genres: List[str] = []
    total_pages: int = 0
    current_page: int = 0
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    notes: str = ""
    languages: List[str] = []
    review: str = ""
    favorite: bool = False
    author_country: Optional[str] = None
    series: Optional[str] = None
    series_position: Optional[int] = None
    original_year: Optional[int] = None
    reading_format: str = "reading"
    last_read: Optional[str] = None

class BookResponse(BaseModel):
    id: UUID
    title: str
    author: str
    status: Optional[str] = None
    rating: Optional[int] = None
    cover: Optional[str] = None
    section: Optional[str] = None
    genres: List[str] = []
    total_pages: int = 0
    current_page: int = 0
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    notes: str = ""
    languages: List[str] = []
    review: str = ""
    favorite: bool = False
    author_country: Optional[str] = None
    series: Optional[str] = None
    series_position: Optional[int] = None
    original_year: Optional[int] = None
    reading_format: str = "reading"
    last_read: Optional[str] = None
    created_at: int

class UserBookResponse(BaseModel):
    id: UUID
    user_id: UUID
    book_id: UUID
    status: str
    book: BookResponse